update to 1.21.7

HongMeiIing opened the pull request:
should work with 1.21.6 and 1.21.8 if compiled for those versions

replaced ParrotEntity.writeNbt with ParrotEntity.writeData as the former seems to have been replaced with the latter
replaced GameVersion.getSaveVersion().getId() with GameVersion.dataVersion().id() for the same reason
replaced ItemStack.fromNbt and ItemStack.toNbt with their implementations (taken from yarn 1.21.5) as fromNbt and toNbt don't seem to exist anymore

Thomilist committed:
Upgrade Gradle wrapper to 8.14.1

Commit:
update to 1.21.7

HongMeiIing commented:
tested with 1.21.8, works

Daniel99j2 commented:
[dimensional-inventories-2.2.1+1.21.7.zip](https://github.com/user-attachments/files/21955086/dimensional-inventories-2.2.1%2B1.21.7.zip)
For those wanting it, here's the built jar's (pick the one with the short name)

Thomilist committed:
Upgrade Gradle wrapper to 8.14.3

Thomilist committed:
Tweaked module registration logic

— Repeated registrations of the same module type now overwrite previous registrations instead of failing.
— Module string formatting moved into the module interface itself.

Thomilist committed:
Streamlined gametest setup and teardown logic

— Moved some boilerplate into the gametest base class.

Thomilist committed:
Added (failing) test to trigger double transition handling bug when respawning across dimension pools

Thomilist committed:
Fixed test ordering bug

— Tests are supposed to run sequentially to ensure mod data is stashed as expected (not mixed across tests).

Thomilist committed:
Fixed double transition handling when respawning across dimension pools

— Fixes #25.

Thomilist committed:
Corrected naming convention inconsistency

— Too much C# at work, oops...

Thomilist committed:
Update to 1.21.6

Thomilist committed:
Merge branch 'develop' into develop

Thomilist commented:
Sorry for the long wait, but thank you very much for the pull request!

I agree with the way you've handled the breaking changes from Minecraft, so even though it may not look like it with the long list of additional commits, I've merged it almost as-is. Most of those commits are changes made for the [2.2.1+1.21.5](https://github.com/Thomilist/dimensional-inventories/releases/tag/2.2.1%2B1.21.5) version, which I've simply included here. Aside from that, it's a slight tweak to the `toItemStack` method (returning `null` on partial/erroneous results to trigger the lost-and-found logic), as well as targeting 1.21.6 on the merged commit to keep the updates in order.

I've posted versions for 1.21.6, 1.21.7 and 1.21.8 now, both [here on GitHub](https://github.com/Thomilist/dimensional-inventories/releases) and on [Modrinth](https://modrinth.com/mod/dimensional-inventories/versions). But yeah, thank you very much :)
